### PR TITLE
Supportability Metrics for log forwarding

### LINKED ISSUE #937: Synchronize supportability metrics for logging
Updated with most recent brief.

The following metrics are new and should be implemented in the AgentHealthReporter.  They must only be sent on connect (once).  Use the standard pattern for implementing these. These will can be checked in our configuration.
- Supportability/Logging/Metrics/{language}/{enabled|disabled}
- Supportability/Logging/Forwarding/{language}/{enabled|disabled}
- Supportability/Logging/LocalDecorating/{language}/{enabled|disabled}

This metrics should be sent once on connect (or close  to it) if a logging frame is being instrumented
- Supportability/Logging/enabled/{language}/{framework}

The following metrics are already implemented and will need adjusted to fit with the new naming scheme.  These metrics are found in the LogEventAggregator. Use the standard pattern for implementing these.

- Supportability/Logging/Forwarding/Seen replaces - Supportability/LoggingEvents/TotalLoggingEventsCollected
- Supportability/Logging/Forwarding/Sent replaces Supportability/LoggingEvents/TotalLoggingEventsSent

Add this metric:
- Supportability/Logging/Forwarding/Dropped

The following metric should be removed:
- Supportability/LoggingEvents/TotalLoggingEventsRecollected

Add two new metrics - different names, same data, in the LogEventAggregator that is the difference between seen and sent.  This will require us to track the sent/seen values in LogEventAggregator or AgentHealthReporter so that we can build the metric.

- Supportability/Logging/Forwarding/Dropped
- Logging/Forwarding/Dropped

This should include unit and integration tests as needed.

### PR DESCRIPTION
**Pull request by JcolemanNR**
## Description

Closes #937 

Renames several supportability metrics, and adds a few new ones for log forwarding. 

Introduced a way to use the agent health reporter to send metrics once on startup, or after any configuration change.

# Author Checklist
- [x] Unit tests, Integration tests, and Unbounded tests completed

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)

### CODE REVIEW

File: src/Agent/NewRelic/Agent/Core/AgentHealth/AgentHealthReporter.cs [hunk 1]
```diff
@@ -108,20 +107,16 @@ private void ReportSupportabilityGaugeMetric(string metricName, float value)
...
-			var metric = _metricBuilder.TryBuildDotnetFrameworkVersionMetric(AgentInstallConfiguration.DotnetFrameworkVersion);
+            var metric = _metricBuilder.TryBuildDotnetFrameworkVersionMetric(AgentInstallConfiguration.DotnetFrameworkVersion);
 #else
             var metric = _metricBuilder.TryBuildDotnetCoreVersionMetric(AgentInstallConfiguration.DotnetCoreVersion);
 #endif
             TrySend(metric);
         }
 
-        public void ReportAgentVersion(string agentVersion, string hostName)
+        public void ReportAgentVersion(string agentVersion)
```
Comment (jaffinito) [inline src/Agent/NewRelic/Agent/Core/AgentHealth/AgentHealthReporter.cs:117]: Nice bit of clean up!

**Review by jaffinito — APPROVED** [review]
Great work!

File: src/Agent/NewRelic/Agent/Core/AgentHealth/AgentHealthReporter.cs [hunk 2]
```diff
@@ -536,6 +531,12 @@ public void ReportSupportabilityCollectorErrorException(string endpointMethod, T
 
         #region Log Events and Metrics
 
+        private ConcurrentDictionary<string, bool> loggingFrameworksReported = new ConcurrentDictionary<string, bool>();
```
Comment (nr-ahemsath) [inline src/Agent/NewRelic/Agent/Core/AgentHealth/AgentHealthReporter.cs:534]: Minor nitpick: our coding style guidelines are to prefix private class members with an underscore, e.g. `_loggingFrameworksReported`

File: src/Agent/NewRelic/Agent/Core/AgentHealth/AgentHealthReporter.cs [hunk 3]
```diff
@@ -562,28 +572,47 @@ public void IncrementLogLinesCount(string logLevel)
...
+
         #endregion
 
         public void ReportSupportabilityPayloadsDroppeDueToMaxPayloadSizeLimit(string endpoint)
         {
             TrySend(_metricBuilder.TryBuildSupportabilityPayloadsDroppedDueToMaxPayloadLimit(endpoint));
         }
 
+        // Only one metric harvest happens at a time, so locking around this bool is not important
+        private bool OneTimeMetricsCollected;
```
Comment (nr-ahemsath) [inline src/Agent/NewRelic/Agent/Core/AgentHealth/AgentHealthReporter.cs:592]: same here: `_oneTimeMetricsCollected`

File: src/Agent/NewRelic/Agent/Core/AgentHealth/AgentHealthReporter.cs [hunk 4]
```diff
@@ -697,6 +726,12 @@ private void CollectSupportabilityDataUsageMetrics()
             }
         }
 
+        protected override void OnConfigurationUpdated(ConfigurationUpdateSource configurationUpdateSource)
```
Comment (nr-ahemsath) [inline src/Agent/NewRelic/Agent/Core/AgentHealth/AgentHealthReporter.cs:729]: Great job figuring out how to report changes!

**Review by nr-ahemsath — CHANGES_REQUESTED** [review]
A few changes requested:
1. A couple of local private variable names that don't fit our existing code style.
2. It would be nice to go ahead and move more supportability metrics like agent version to the one-time metrics.

**Review by nr-ahemsath — APPROVED** [review]
Approved! 👍

**Pull request merged**